### PR TITLE
Update the `AboutCard` to properly support non-standard entity types and rework the defaults for the build-in kinds

### PR DESCRIPTION
**Pull request by dhenneke**
Some recommended fields were not displayed correctly (e.g. `System` didn't tell "No System" but was just blank if missing). This change also hides all fields that are not available in case the entity has a org specific kind that is not known by the upstream backstage. And I added some tests for the `AboutContent` to test the most common entity types.

I also changed the <kbd>✏️ Edit</kbd> button in the `AboutCard` to a proper backstage `Link` instead of using `window.open(...)`. I prefer to see the URL on hover instead of magically get a new window opened.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx [hunk 1]
```diff
@@ -104,14 +121,22 @@ export const AboutContent = ({ entity }: Props) => {
           />
         </AboutField>
       )}
-      {!isSystem && !isDomain && (
+      {(isAPI ||
+        isComponent ||
+        isResource ||
+        isTemplate ||
+        isGroup ||
+        isLocation ||
```
[the remark this answers is not in the document]
Comment (dhenneke) [reply plugins/catalog/src/components/AboutCard/AboutContent.tsx:129]: Considering that `type` and `location` are required fields and missing values shouldn't be allowed by the catalog validation, we could simply change it to "display if available".

The most value of this "No ..." fallback is in the optional fields "system", "domain", and "tags" to show that this feature is available and might be recommended. Even though a user still doesn't receive any help on how to set these values. An option could be to add a hover text that says what to do in this case.

Depends on how opinionated we want to be about the data model in the default `AboutCard`.

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog/src/components/AboutCard/AboutContent.tsx:129]: Yep if we want to add even more help text and special behaviors then it might perhaps be worth doing it your way.

The main thing I have against it, is that it makes it less usable for custom kinds. We expect companies to expand the model with their own kinds, and it's maybe convenient to be able to use the unmodified about box for them too. Honestly, this is kinda arguable, another way to look at it is that the superior approach is to make it very easy to compose a custom about box in any way you like.

In summary I don't have a strong opinion here :) The only thing that held back my approve was actually the question about Link targets above.

File: plugins/catalog/src/components/AboutCard/AboutCard.tsx [hunk 2]
```diff
@@ -148,12 +148,12 @@ export function AboutCard({ variant }: AboutCardProps) {
...
           <IconButton
+            component={Link}
             aria-label="Edit"
             disabled={!entityMetadataEditUrl}
             title="Edit Metadata"
-            onClick={() => {
-              window.open(entityMetadataEditUrl ?? '#', '_blank');
-            }}
+            to={entityMetadataEditUrl ?? '#'}
+            target="_blank"
```
[the remark this answers is not in the document]
Comment (dhenneke) [reply plugins/catalog/src/components/AboutCard/AboutCard.tsx:156]: fixed it 👍

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx [hunk 3]
```diff
@@ -40,9 +40,13 @@ type Props = {
 export const AboutContent = ({ entity }: Props) => {
   const classes = useStyles();
   const isSystem = entity.kind.toLocaleLowerCase('en-US') === 'system';
-  const isDomain = entity.kind.toLocaleLowerCase('en-US') === 'domain';
   const isResource = entity.kind.toLocaleLowerCase('en-US') === 'resource';
   const isComponent = entity.kind.toLocaleLowerCase('en-US') === 'component';
+  const isAPI = entity.kind.toLocaleLowerCase('en-US') === 'api';
+  const isTemplate = entity.kind.toLocaleLowerCase('en-US') === 'template';
+  const isLocation = entity.kind.toLocaleLowerCase('en-US') === 'location';
+  const isGroup = entity.kind.toLocaleLowerCase('en-US') === 'group';
```
[the remark this answers is not in the document]
Comment (dhenneke) [reply plugins/catalog/src/components/AboutCard/AboutContent.tsx:48]: There is nothing special about these two :-)

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhenneke) [reply plugins/catalog/src/components/AboutCard/AboutContent.tsx:129]: The logic here is that it now always shows the field if it is available and hide them if not. But for certain selected kinds,  it also shows whether a field is _not_ available. But this is now a positive check and not a negative check that falls back to show every field for any unrecognized entity kind.

Lets go with this for now and see if someone has more ideas in the future :-)

**Review by freben — APPROVED** [review]
nice!
